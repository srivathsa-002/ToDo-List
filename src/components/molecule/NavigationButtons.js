import React from 'react';
import { Badge } from "@material-ui/core";
import AllIcon from "../atoms/AllIcon";
import ActiveIcon from "../atoms/ActiveIcon";
import CompleteIcon from "../atoms/CompleteIcon";
import "../../styles/components/container.css";

export default class NavigationButtons extends React.Component {
    render() {
        return (
            <div className="button-container" data-testid="navigationButtons">
                <div className="three-buttons">
                    <div className="hoverable1" onClick={this.props.all} val="All">
                        <AllIcon />
                        {" "}All
                </div>
                    <hr />
                    <div className="hoverable2" onClick={this.props.active} val="Active">
                        <ActiveIcon />
                        <Badge badgeContent={this.props.badgeContent} color="primary">
                            {" "}Active
                  </Badge>
                    </div>
                    <hr />
                    <div className="hoverable3" onClick={this.props.complete} val="Complete">
                        <CompleteIcon />
                        {" "}Completed
                  </div>
                </div>
            </div>
        );
    }
}