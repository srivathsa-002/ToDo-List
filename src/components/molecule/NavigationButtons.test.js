import React from "react";
import {render,fireEvent} from "@testing-library/react";
import NavigationButtons from "./NavigationButtons";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("NavigationButtons testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<NavigationButtons />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly <NavigationButtons>", () => {
    const{getByTestId} = render(<NavigationButtons />)

    expect(getByTestId("navigationButtons")).toBeInTheDocument()
})