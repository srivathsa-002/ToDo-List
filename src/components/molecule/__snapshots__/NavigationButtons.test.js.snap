// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButtons testing Matches the snapshot 1`] = `
<div
  className="button-container"
  data-testid="navigationButtons"
>
  <div
    className="three-buttons"
  >
    <div
      className="hoverable1"
      val="All"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-testid="all"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
        />
      </svg>
       
      All
    </div>
    <hr />
    <div
      className="hoverable2"
      val="Active"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-testid="active"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
      <span
        className="MuiBadge-root"
      >
         
        Active
        <span
          className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
        />
      </span>
    </div>
    <hr />
    <div
      className="hoverable3"
      val="Complete"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-testid="complete"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 10H2v2h12v-2zm0-4H2v2h12V6zM2 16h8v-2H2v2zm19.5-4.5L23 13l-6.99 7-4.51-4.5L13 14l3.01 3 5.49-5.5z"
        />
      </svg>
       
      Completed
    </div>
  </div>
</div>
`;
